### PR TITLE
Add rule only if filled and add message

### PR DESCRIPTION
**Pull request by vysinsky**
Now it throws error 'PHP User Warning: Missing validation message for control '...' when field is not filled.

### CODE REVIEW

**Comment by vysinsky** [comment]
ping @vojtech-dobes :)

**Comment by vojtech-dobes** [comment]
:)

Isn't problem that error message is hardcoded?

File: src/CalendarPicker.php [hunk 1]
```diff
@@ -79,16 +82,27 @@ class CalendarPicker extends BaseControl
...
 	 * @param  string|NULL
+	 * @param  string
+	 * @param  string
 	 */
-	public function __construct($label = NULL)
+	public function __construct($label = NULL, $invalidDateMessage = self::DEFAULT_INVALID_DATE_MESSAGE, $invalidTimeMessage = self::DEFAULT_INVALID_TIME_MESSAGE)
 	{
 		parent::__construct($label);
+		$this->invalidDateMessage = $invalidDateMessage;
+		$this->invalidTimeMessage = $invalidTimeMessage;
```
Comment (vojtech-dobes) [inline src/CalendarPicker.php]: invalidTimeMessage can be part of `useTime`

Comment (vysinsky) [inline src/CalendarPicker.php]: yeah, true

**Pull request merged**